src/middleware.ts

```diff
@@ -5,9 +5,15 @@ import { getToken } from "next-auth/jwt";
 export async function middleware(req: NextRequest) {
   const token = await getToken({ req });
   const isAuth = !!token;
+  const isSignedUp =
+    (token?.profile as { firstName: string | null })?.firstName !== null &&
+    (token?.profile as { lastName: string | null })?.lastName !== null;
+
   const fromUrl = req.nextUrl.pathname;
   const isAuthPage =
     fromUrl.startsWith("/auth/login") || fromUrl.startsWith("/auth/signup");
+  const isNamePage = fromUrl.startsWith("/onboard/name");
+  const isMemberPage = fromUrl.startsWith("/onboard/member");
```

Comment:
instead of checking both cases you can just check if it starts with `"/onboard"`